integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/message/processor/test/DeactivatedCappMPBehaviourOnRestartTestCase.java

```diff
@@ -80,15 +68,13 @@ public void testDeactivatedMPUponServerRestart() throws Exception {
         axis2Server.start();
 
         // Restart the ESB Server after the MP is deactivated.
-        serverConfigurationManager.restartGracefully();
+        CarbonServerExtension.restartServer();
 
         super.init();
-        logViewerClient = new LogViewerClient(contextUrls.getBackEndUrl(), getSessionCookie());
+        CarbonLogReader carbonLogReader = new CarbonLogReader();
```

Comment:
If we define the log reader  before server start and start it , we might get the expected logs.

i.e before line https://github.com/wso2/micro-integrator/pull/336/files#diff-a1a50aa3c0dc71fac52b25f6529d8a38R71

Reply:
Fixed.